core/domain/translation_domain.py

```diff
@@ -18,9 +18,258 @@
 
 from __future__ import annotations
 
+import copy
+
 from core import utils
 
-from typing import Dict
+from typing import Dict, List
+from typing_extensions import TypedDict
+
+
+TRANSLATABLE_CONTENT_FORMAT_HTML = 'html'
+TRANSLATABLE_CONTENT_FORMAT_UNICODE_STRING = 'unicode'
+TRANSLATABLE_CONTENT_FORMAT_SET_OF_NORMALIZED_STRING = (
+    'set_of_normalized_string')
+TRANSLATABLE_CONTENT_FORMAT_SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class TranslatableContentDict(TypedDict):
+    """Dictionary representing TranslatableContent object."""
+
+    content_id: str
+    content: str|List[str]
+    content_type: str
+
+
+class TranslatedContentDict(TypedDict):
+    """Dictionary representing TranslatedContent object."""
+
+    content: str|List[str]
+    needs_update: bool
+
+
+class BaseTranslatableObject:
+    """Base class for all translatable objects."""
```

Comment:
Explain what an "object" means here. (The term is vague and could refer to anything -- a dict? an entity? a domain object? etc.) Perhaps provide examples and constraints.